src/lib/annotations/drawing/DrawingController.js

```diff
@@ -0,0 +1,303 @@
+     * @return {void}
+     */
+    handleSelection(event) {
+        if (!event) {
+            return;
+        }
+
+        const location = this.annotator.getLocationFromEvent(event, TYPES.point);
+        if (!location) {
+            return;
+        }
+
+        const eventBoundary = {
+            minX: +location.x - DRAW_BORDER_OFFSET,
+            minY: +location.y - DRAW_BORDER_OFFSET,
+            maxX: +location.x + DRAW_BORDER_OFFSET,
+            maxY: +location.y + DRAW_BORDER_OFFSET
+        };
+
+        // Get the threads that correspond to the point that was clicked on
+        const intersectingThreads = this.threads
+            .search(eventBoundary)
+            .filter((drawingThread) => drawingThread.location.page === location.page);
+
+        // Clear boundary on previously selected thread
+        if (this.selected) {
+            const canvas = this.selected.drawingContext.canvas;
+            this.selected.drawingContext.clearRect(0, 0, canvas.width, canvas.height);
+        }
+
+        // Selected a region with no drawing threads, remove the reference to the previously selected thread
+        if (intersectingThreads.length === 0) {
+            this.selected = undefined;
+            return;
+        }
+
+        // Randomly select a thread in case there are multiple
+        const index = Math.floor(Math.random() * intersectingThreads.length);
+        const selected = intersectingThreads[index];
+        this.select(selected);
+    }
+
+    /**
+     * Select the indicated drawing thread. Deletes a drawing thread upon the second consecutive selection
+     *
+     * @private
+     * @param {DrawingThread} selectedDrawingThread - The drawing thread to select
+     * @return {void}
+     */
+    select(selectedDrawingThread) {
+        if (this.selected && this.selected === selectedDrawingThread) {
+            // Selected the same thread twice, delete the thread
+            const toDelete = this.selected;
+            toDelete.deleteThread();
+
+            // Redraw any threads that the deleted thread could have been covering
+            const toRedraw = this.threads.search(toDelete);
+            toRedraw.forEach((drawingThread) => drawingThread.show());
+            this.selected = undefined;
+        } else {
+            // Selected the thread for the first time, select the thread (TODO @minhnguyen: show UI on select)
+            selectedDrawingThread.drawBoundary();
+            this.selected = selectedDrawingThread;
+        }
+    }
+
+    /**
+     * Toggle the undo and redo buttons based on the number of actions available
+     *
+     * @private
+     * @param {number} undoCount - The number of objects that can be undone
+     * @param {number} redoCount - The number of objects that can be redone
+     * @return {void}
+     */
+    updateUndoRedoButtonEls(undoCount, redoCount) {
+        if (this.undoButtonEl) {
+            if (undoCount === 1) {
```

Comment:
Can these numbers be > 1?

Reply:
Yes, in which case we don't need to do anything because they should already be enabled. In theory, the undo and redo counts will always start at 0. When they hit 1 they will be enabled and stay enabled for values > 1. If they drop to 0 they will return to the starting state (of being disabled).